packages/lesswrong/lib/collections/votes/fragments.ts

```diff
@@ -24,3 +24,17 @@ registerFragment(`
     }
   }
 `)
+
+registerFragment(`
+  fragment UserVotes on Vote {
+    _id
+    userId
+    voteType
+    power
+    cancelled
+    documentId
+    votedAt
+    isUnvote
```

Comment:
Code smell here: What are `cancelled` and `isUnvote` doing in a fragment? Those fields are a sort of soft-delete mechanism, so the only way they can get exposed on the client is if the client is getting cancelled votes.

Reply:
Cancelled was necessary to filter out cancelled votes. (I specifically ran into that problem while testing it)

I think I initially assumed I'd also have to filter out isUnvote, but it looks like I didn't get around to doing that, and I'm not 100% sure what it even means actually.

Reply:
Filtering out cancelled votes should be happening server-side in the view, before the fragment becomes relevant. (Which I think is already what's happening.)

Reply:
It wasn't happening, at least the way I was fetching the votes. (I've updated that this may require an overall different approach than I was doing)

It looks like currently most of the way deal with votes is through "currentUserVote" on makeVoteable, which is where the "cancelled-filter" lives.